Overhaul of example

WardLT opened the pull request:
- Stripped out unnecessary code
- Added a simple environment file
- Added documentation for how to run the code
- Made it work on GPU without removing IPU code
- Added a baseline model ran on GPU
- Modified GNN architecture to improve accuracy by ~3x

WardLT committed:
Removed unnecessary files

WardLT committed:
Added a new environment file, reduced set of moldesign

WardLT committed:
Moved training data to a qm9 folder

Make room for more datasets later

WardLT committed:
Bug fix: remove ref to chemistry module

WardLT committed:
Add nfp and networkx to environment

WardLT committed:
Removed unnecessary notebook

WardLT committed:
Update with the best architecture from manual search

WardLT committed:
Added baselin emodel trained on a single GPU

WardLT committed:
Switch to a requirements file

vksastry commented on run_test.py:
```diff
@@ -72,78 +87,108 @@ def build_fn(atom_features: int = 64, message_steps: int = 8,
+                cb.CSVLogger(test_dir / 'train_log.csv'),
                 cb.TerminateOnNaN()
-            ], steps_per_epoch=10
+            ],
```
steps_per_epoch is required for running on IPU. Otherwise all other changes looks ok. I am merging them to another branch and if it works fine on IPU- then we intend to merge the changes to the master.

WardLT replied:
Sounds like a great plan. Sorry, I should have asked about the "steps_per_epoch" part as it seemed intentional?

Would you like me to detect the number of steps rather than keeping it fixed at a small number?

vksastry replied:
yes, predetermining is the right way to do. Just to test that out initially, we use a fixed number.

WardLT replied:
Makes sense :) I'll get you that change soon

WardLT committed:
Fix the steps per epoch

- Determine the steps based on dataset and batch size
- Use an infinitely-repeating iterator because TF will exhaust loader
otherwise

WardLT commented:
Alright. Fixed the steps per epoch. 🤞 everything works!

vksastry commented:
Merged into develop branch
